var/spack/repos/builtin/packages/thrift/package.py

```diff
@@ -34,78 +35,123 @@ class Thrift(Package):
     version("0.10.0", sha256="2289d02de6e8db04cbbabb921aeb62bfe3098c4c83f36eec6c31194301efa10b")
     version("0.9.3", sha256="b0740a070ac09adde04d43e852ce4c320564a292f26521c46b78e0641564969e")
 
-    variant("pic", default=True, description="Build position independent code")
-    variant("c", default=True, description="Build support for C-family languages")
+    variant("openssl", default=False, description="Build with OpenSSL")
+    variant("cpp", default=True, description="Build C++ library")
+    with when("+cpp"):
+        variant("shared", default=True, description="Build shared libraries")
+        variant("libevent", default=False, description="Build with libevent support")
+        variant("qt5", default=False, description="Build with Qt5 support")
+        variant("zlib", default=False, description="Build with ZLIB support")
+    variant("c_glib", default=False, description="Build C (GLib) library")
     variant("java", default=False, description="Build support for java")
+    variant("javascript", default=False, description="Build Javascript library")
+    variant("nodejs", default=False, description="Build NodeJS library")
     variant("python", default=True, description="Build support for python")
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
-
-    depends_on("pkgconfig", type="build")
-    depends_on("autoconf", type="build")
-    depends_on("automake", type="build")
-    depends_on("libtool", type="build")
-    depends_on("boost@1.53:")
-
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
-    depends_on("bison", type="build")
-    depends_on("flex", type="build")
-    depends_on("openssl")
-
-    # Variant dependencies
-    depends_on("zlib-api", when="+c")
-    depends_on("libevent", when="+c")
-
-    depends_on("java@7:", when="+java")
-    depends_on("ant", when="+java")
-
-    extends("python", when="+python")
-    depends_on("python@:3.11.9", when="+python")
-    depends_on("py-setuptools", type=("build", "run"), when="+python")
-    depends_on("py-six@1.7.2:", type=("build", "run"), when="@0.10.0:+python")
-    depends_on("py-tornado", type=("build", "run"), when="+python")
-    depends_on("py-twisted", type=("build", "run"), when="+python")
-    depends_on("py-zope-interface", type=("build", "run"), when="+python")
-    depends_on("py-pure-sasl", type=("build", "run"), when="+python")
-    depends_on("scons", type=("build", "run"), when="+python")
-    depends_on("gmake", type="build")
+    build_system("cmake", "autotools", default="autotools")
+
+    with default_args(type="build"):
+        depends_on("c")
+        depends_on("cxx")
```

Comment:
I think C is only required when `+c`, but C++ is definitely always required, even when `~cpp`